UploadBehavior.php

```diff
@@ -197,9 +197,8 @@ public function afterSave()
      */
     public function beforeDelete()
     {
-        $attribute = $this->attribute;
-        if ($this->unlinkOnDelete && $attribute) {
-            $this->delete($attribute);
+        if ($this->unlinkOnDelete) {
+            $this->deleteFolder($this->path);
```

Comment:
The directory can contain a few files. I do not understand where you check that this directory is empty?